FWIO/RNTuple/plugins/RNTupleOutputFile.cc

```diff
@@ -0,0 +1,583 @@
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt32}});
+      } else if (typeName == "std::int64_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt64}});
+      } else if (typeName == "float") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal32}});
+      } else if (typeName == "double") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal64}});
+      }
+    }
+
+    void checkSubFieldsForNoSplit(ROOT::Experimental::RFieldBase& iField,
+                                  std::vector<std::string> const& iNoSplitFields) {
+      for (auto const& name : iNoSplitFields) {
+        if (name.starts_with(iField.GetFieldName())) {
+          bool found = false;
+          for (auto& subfield : iField) {
+            if (subfield.GetQualifiedFieldName() == name) {
+              found = true;
+              noSplitField(subfield);
+              break;
+            }
+          }
+          if (not found) {
+            edm::LogError("MissingSubField")
+                << "The data product was found but the requested subfield '" << name << "' is not part of the class";
+          }
+        }
+      }
+    }
+  }  // namespace
+
+  void RNTupleOutputFile::setupDataProducts(SelectedProducts const& iProducts,
+                                            std::vector<bool> const& iUseStreamer,
+                                            std::vector<std::string> const& iNoSplitFields,
+                                            RNTupleModel& iModel) {
+    unsigned int index = 0;
+    const bool noSplitSubFields = (iNoSplitFields.size() == 1 and iNoSplitFields[0] == "all") ? true : false;
+    for (auto const& prod : iProducts) {
+      try {
+        edm::convertException::wrap([&]() {
+          if (index >= iUseStreamer.size() or not iUseStreamer[index]) {
+            auto field = ROOT::Experimental::RFieldBase::Create(fixBranchName(prod.first->branchName()),
+                                                                prod.first->wrappedName())
+                             .Unwrap();
+            if (noSplitSubFields) {
+              //use the 'conventional' way to store fields
+              for (auto& subfield : *field) {
+                noSplitField(subfield);
+              }
+            } else if (not iNoSplitFields.empty()) {
+              checkSubFieldsForNoSplit(*field, iNoSplitFields);
+            }
+            iModel.AddField(std::move(field));
+            branchesWithStoredHistory_.insert(prod.first->branchID());
+          } else {
+            auto field = std::make_unique<ROOT::Experimental::RStreamerField>(fixBranchName(prod.first->branchName()),
+                                                                              prod.first->wrappedName());
+            iModel.AddField(std::move(field));
+            branchesWithStoredHistory_.insert(prod.first->branchID());
```

Comment:
It seems to me this line could be moved out from the if-else blocks, as it is the last statement in both cases.

Reply:
done